### PR TITLE
Remove binary .jar files in tests

### PR DESCRIPTION
**Pull request by flodet-me**
## Description

Using `JarCreator` to create .jar files during tests instead of binary files in the repository.

Simple solution for just this module.

## Checklist

* [X] I have updated the documentation accordingly.
* [X] I have added tests to cover my changes.
* [X] All new and existing tests passed.
* [X] My code follows the code style of this project.
* [X] I ran all checks which produced no new errors or warnings for my changes.
* [X] I have checked to ensure there aren't other open Pull Requests for the same update/change.

### CODE REVIEW

File: sapl-pdp-embedded/src/test/java/io/sapl/util/JarCreator.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.jar.JarEntry;
+import java.util.jar.JarOutputStream;
+import java.util.stream.Stream;
+
+/**
+ * Utility class for creating plain JAR archives.
+ */
+public final class JarCreator {
```
Comment (heutelbeck) [inline sapl-pdp-embedded/src/test/java/io/sapl/util/JarCreator.java:35]: Reduce keywords and add private constructor.
Use ```@UtilityClass``` see Lombok documentation.

File: sapl-pdp-embedded/src/test/java/io/sapl/util/JarCreator.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+            for (String sourcePath : sourcePaths) {
+                var source = Path.of(sourcePath);
+                addFileToJar(source, source, jos);
+            }
+        }
+    }
+
+    private static URL createJarFromResource(String jarName, String resourcePath, String jarFolderReference)
+            throws IOException {
+        var path = System.getProperty("java.io.tmpdir") + "/" + jarName;
```
Comment (heutelbeck) [inline sapl-pdp-embedded/src/test/java/io/sapl/util/JarCreator.java:59]: Please use the JUnit tools for tmpdir support:
E.g., see: https://www.baeldung.com/junit-5-temporary-directory#tmpdir-extension-1
Double check, that die directory is deleted after the tests conclude.

**Review by heutelbeck — CHANGES_REQUESTED** [review]

File: sapl-pdp-embedded/src/test/java/io/sapl/util/JarCreator.java (hunk 2 above)
Comment (flodet-me) [reply sapl-pdp-embedded/src/test/java/io/sapl/util/JarCreator.java:59]: Resolving by forwarding the tempdir from the Test method property injection and setting the `CleanupMode.Always`.

**Comment by flodet-me** [comment]
> ## [Quality Gate Failed](https://sonarcloud.io/dashboard?id=heutelbeck_sapl-policy-engine&pullRequest=92) **Quality Gate failed**
> 
> Failed conditions
> 
> [C Reliability Rating on New Code](https://sonarcloud.io/dashboard?id=heutelbeck_sapl-policy-engine&pullRequest=92) (required ≥ A)
> 
> [See analysis details on SonarCloud](https://sonarcloud.io/dashboard?id=heutelbeck_sapl-policy-engine&pullRequest=92)
> 
> idea Catch issues before they fail your Quality Gate with our IDE extension SonarLint [SonarLint](https://www.sonarsource.com/products/sonarlint/features/connected-mode/?referrer=pull-request)

I think this is not 100% working on just my changes, as most changes are based on different methods, that did not change.

**Review by heutelbeck — APPROVED** [review]

**Pull request merged**